js/controllers/todos_controller.js

```diff
@@ -1,4 +1,125 @@
-App.TodosController = Ember.ObjectController.extend({
+
+Abs.TasksController = Ember.ObjectController.extend({
     actions: {
+        editTasks: function () {
+            this.set('editMode', true);
+        },
+        changeTask: function () {
+            var me = this;
+            Ember.run.later((function () {
+                me.set('editMode', false);
+                if (Ember.isEmpty(me.get('model.title'))) {
+                    me.send('removeTask');
+                } else {
+                    me.get('model').save();
+                }
+            }), 100);
+        },
+        removeTask: function () {
+            var Task = this.get('model');
+            Task.deleteRecord();
+            Task.save();
+        }
+    },
+
+    editMode: false,
+
+    isDone: function (key, value) {
+        var model = this.get('model');
+
+        if (value === undefined) {
+            // property being used as a getter
+            return model.get('isDone');
+        } else {
+            // property being used as  setter
+            model.set('isDone', value);
+            model.save();
+            return value;
+        }
+    }.property('model.isDone')
+});
+
+Abs.AbsController = Ember.ArrayController.extend({
+    actions: {
+        checkAll: function () {
+            var checked = $('#chk-all')[0].checked;
```

Comment:
"#chk-all" is my selector to get the select all checkbox, so I want read this value and change other checkbox value to it, and [0] is for get the first element by this id, its better to change to a class  


     On Monday, September 28, 2015 5:22 PM, Ruan Naude <notifications@github.com> wrote:
   

 In js/controllers/todos_controller.js:> +        if (value === undefined) {
> +            // property being used as a getter
> +            return model.get('isDone');
> +        } else {
> +            // property being used as  setter
> +            model.set('isDone', value);
> +            model.save();
> +            return value;
> +        }
> +    }.property('model.isDone')
> +});
> +
> +Abs.AbsController = Ember.ArrayController.extend({
> +    actions: {
> +        checkAll: function () {
> +            var checked = $('#chk-all')[0].checked;
why do you do $('#chk-all')[0]?—
Reply to this email directly or view it on GitHub.